Source/GoBot/ViewDatabase.swift

```diff
@@ -218,6 +230,8 @@ class ViewDatabase {
         
         self.currentUser = user
         self.currentUserID = try self.authorID(of: user, make: true)
+
+        try setAllMessagesAsReadIfNeeded(on: db)
```

Comment:
Why did you choose to call this from outside the migration code? In my mind it could just be called down in the `if db.userVersion == 13 {`. That would let us get rid of the `needsToSetAllMessagesAsRead` state variable.

Reply:
I did it that way because during migrations we don't have the currentUserId yet. I could assume it is 1 or I could move the line that do the migrations two steps forward so we have the user id or I could do it the way i did it. I chose the latter because it was harmless, I don't know what could go wrong with doing a simple query before migrating, but it doesn't look clean, maybe in the future we need to do something before any query is done.

Reply:
Ok that makes sense. I don't think it would be too bad to do a read during the migration but also I think it's fine the way it is.